process/economics/rewardsPerBlock.go

```diff
@@ -0,0 +1,233 @@
+	}
+
+	noncePerShardPrevEpoch, prevEpochStart, err := r.startNoncePerShardFromPreviousEpochStart(metaBlock.Epoch - 1)
+	if err != nil {
+		return nil, err
+	}
+	prevEpochEconomics := prevEpochStart.EpochStart.Economics
+
+	noncePerShardCurrEpoch, err := r.startNoncePerShardFromLastCrossNotarized(metaBlock.GetNonce(), metaBlock.EpochStart)
+	if err != nil {
+		return nil, err
+	}
+
+	roundsPassedInEpoch := metaBlock.GetRound() - prevEpochStart.GetRound()
+	maxBlocksInEpoch := roundsPassedInEpoch * uint64(r.shardCoordinator.NumberOfShards())
+	totalNumBlocksInEpoch := r.computeNumOfTotalCreatedBlocks(noncePerShardPrevEpoch, noncePerShardCurrEpoch)
+
+	inflationRate, err := r.computeInflationRate(prevEpochEconomics.TotalSupply, prevEpochEconomics.NodePrice)
+	if err != nil {
+		return nil, err
+	}
+
+	rwdPerBlock := r.computeRewardsPerBlock(prevEpochEconomics.TotalSupply, maxBlocksInEpoch, inflationRate)
+	totalRewardsToBeDistributed := big.NewInt(0).Mul(rwdPerBlock, big.NewInt(0).SetUint64(totalNumBlocksInEpoch))
+
+	newTokens := big.NewInt(0).Sub(totalRewardsToBeDistributed, metaBlock.AccumulatedFeesInEpoch)
+	if newTokens.Cmp(big.NewInt(0)) < 0 {
+		newTokens = big.NewInt(0)
+		totalRewardsToBeDistributed = big.NewInt(0).Set(metaBlock.AccumulatedFeesInEpoch)
+		rwdPerBlock = big.NewInt(0).Div(totalRewardsToBeDistributed, big.NewInt(0).SetUint64(totalNumBlocksInEpoch))
+	}
+
+	computedEconomics := block.Economics{
+		TotalSupply:            r.computeRewardsPerValidatorPerBlock(rwdPerBlock),
+		TotalToDistribute:      big.NewInt(0).Add(prevEpochEconomics.TotalSupply, newTokens),
+		TotalNewlyMinted:       big.NewInt(0).Set(newTokens),
+		RewardsPerBlockPerNode: big.NewInt(0).Set(totalRewardsToBeDistributed),
+		// TODO: get actual nodePrice from auction smart contract (currently on another feature branch, and not all features enabled)
+		NodePrice: big.NewInt(0).Set(prevEpochEconomics.NodePrice),
+	}
+
+	return &computedEconomics, nil
+}
+
+// compute rewards per node per block
+func (r *rewardsPerBlock) computeRewardsPerValidatorPerBlock(rwdPerBlock *big.Int) *big.Int {
+	numOfNodes := r.nodesCoordinator.GetNumTotalEligible()
+	return big.NewInt(0).Div(rwdPerBlock, big.NewInt(0).SetUint64(numOfNodes))
+}
+
+// compute inflation rate from totalSupply and totalStaked
+func (r *rewardsPerBlock) computeInflationRate(_ *big.Int, _ *big.Int) (float64, error) {
+	//TODO: use prevTotalSupply and nodePrice (number of eligible + number of waiting)
+	// for epoch which ends now to compute inflation rate according to formula provided by L.
+	return r.rewardsHandler.MaxInflationRate(), nil
+}
+
+const numberOfDaysInYear = 365.0
+
+// compute rewards per block from according to inflation rate and total supply from previous block and maxBlocksPerEpoch
+func (r *rewardsPerBlock) computeRewardsPerBlock(
+	prevTotalSupply *big.Int,
+	maxBlocksInEpoch uint64,
+	inflationRate float64,
+) *big.Int {
+	inflationRatePerDay := inflationRate / numberOfDaysInYear
```

Comment:
We assume that 1day == 1 epoch
Should consider the roundTime, how many rounds are in an year and how many rounds where in the epoch that just ended. In this way the reward will be computed correctly even in the case of early end of epoch event.

Reply:
done.